Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/storage/ADLSGen2StorageAccount.java

```diff
@@ -13,4 +27,37 @@ public ADLSGen2StorageAccount(IClusterDetail clusterDetail, String fullStorageBl
     public StorageAccountTypeEnum getAccountType() {
         return StorageAccountTypeEnum.ADLSGen2;
     }
+
+    public Observable<AccessKey> getAccessKeyList(SubscriptionDetail subscription) {
+        try {
+            Azure azure = AuthMethodManager.getInstance().getAzureManager().getAzure(subscription.getSubscriptionId());
+            return azure.storageAccounts().listAsync()
+                    .filter(ac -> ac.name().equals(getName()))
+                    .map(ac -> ac.resourceGroupName())
+                    .firstOrDefault("")
+                    .doOnNext(rg -> {
+                        if (rg.isEmpty()) {
+                            throw new IllegalArgumentException("Cannot get resource group according to storage account.");
+                        }
+                    })
+                    .map(rg -> new HttpPost(String.format("https://management.azure.com/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Storage/storageAccounts/%s/listKeys",
+                            subscription.getSubscriptionId(), rg, getName())))
+                    .flatMap(post -> new AzureHttpObservable(subscription, "2018-07-01").request(post, null, null, null))
+                    .map(resp -> {
+                        try {
+                            Optional<GetAccessKeyListResponse> keys = ObjectConvertUtils.convertEntityToObject(resp.getEntity(), GetAccessKeyListResponse.class);
+                            if (keys.isPresent()) {
+                                return keys.get().getKeys();
+                            } else {
+                                throw new IllegalArgumentException("Can't get valid access key from response");
+                            }
+                        } catch (IOException ex) {
+                            throw new IllegalArgumentException("Can't create shared key credential.Please check key validation");
```

Comment:
Why needs this?

Reply:
ObjectConvertUtils.convertEntityToObject throws IOException

---
In reply to: [274747286](https://github.com/Microsoft/azure-tools-for-java/pull/2961#discussion_r274747286) [](ancestors = 274747286)